test/libdnf/system/test_state.cpp

```diff
@@ -63,6 +63,16 @@ enabled_stream = "stream-1"
 
 )"""};
 
+static inline void erase_newlines(std::string & str) {
+    str.erase(std::remove(str.begin(), str.end(), '\n'), str.cend());
+}
+
+static void assert_equal_without_newlines(std::string expected, std::string actual) {
+    erase_newlines(expected);
+    erase_newlines(actual);
```

Comment:
You only really need to remove the trailing newline from the stored tomls, not from the static strings that we compare against (or store in the test files).